### ISSUE TITLE
Documentation of MessageObject "deprecated" property

### ISSUE THREAD

**Issue by c-pius** [asyncapi/asyncapi#477]
**Describe the bug**

The schema of the MessageObject contains a [deprecated](https://github.com/asyncapi/asyncapi/blob/5ad3a971c82b33717fdec4c8d78173294ae253ff/versions/2.0.0/schema.json#L686) property. This property is not included in the [MessageObject documentation](https://www.asyncapi.com/docs/specifications/2.0.0#messageObject) or the [markdown](https://github.com/asyncapi/asyncapi/blob/master/versions/2.0.0/asyncapi.md#message-object). Shouldn't it be though?

**Comment by jonaslagoni** [comment]
Good catch @c-pius, can see the [parser](https://github.com/asyncapi/parser-js/blob/master/API.md) does not have the property as well. Tagging @fmvilas to get his perspective, think this is something that was added early on.

**Comment by jonaslagoni** [comment]
I can see that `messageTrait` and`channelItem` also contains a deprecated property which are not present in the tools.
https://github.com/asyncapi/asyncapi/blob/5ad3a971c82b33717fdec4c8d78173294ae253ff/versions/2.0.0/schema.json#L909
https://github.com/asyncapi/asyncapi/blob/5ad3a971c82b33717fdec4c8d78173294ae253ff/versions/2.0.0/schema.json#L501

**Comment by jonaslagoni** [comment]
Related issue about introducing deprecation/sunset property for the channel - https://github.com/asyncapi/asyncapi/issues/305

**Comment by derberg** [comment]
I don't think the `deprecated` flag is supported on a Message or a Channel level. At least I know about Channel -> https://github.com/asyncapi/html-template/pull/81#discussion_r498774249 but don't remember where from 😅 

It might be just a bug in the schema. Remember that schema != spec md file

But yeah, @fmvilas should confirm

**Comment by fmvilas** [comment]
Yeah, I think that's a bug. Always follow the Markdown version of the spec. However, if you think that it still makes sense to have a `deprecated` property at MessageObject level, then let's open an issue to propose it.

**Comment by c-pius** [comment]
Alright, thanks for the info. I also just encountered the `deprecated` property by chance. As of now our use case doesn't require a `deprecated` property on message level. But I could imagine this in the future (maybe even with some more information like successor message, sunset date, etc). If so, I would think about an own extension or proposing something for the MessageObject.

For now you can close this issue from my side.

**Comment by fmvilas** [comment]
Yes, there's a feature request for this (although at channel level): #305. Leaving this open because it needs to be fixed. Thanks!
